tests/api/hello_with_types/src/hello/addon.cpp

```diff
@@ -0,0 +1,62 @@
+/**
+ * @file addon.cpp
+ * @brief A quick 'hello world' Napi Addon in C++
+*/
+
+// Required header and C++ flag
+#if __has_include(<napi.h>) && BUILDING_NODE_EXTENSION
+
+#include <napi.h>
+
+#ifndef STRINGIFY
+# define STRINGIFY_HELPER(n) #n
+# define STRINGIFY(n) STRINGIFY_HELPER(n)
+#endif
+
+namespace Napi
+{
+namespace NAPI_CPP_CUSTOM_NAMESPACE
+{
+
+Napi::Value Hello(const Napi::CallbackInfo& info) {
+  return Napi::String::New(info.Env(), STRINGIFY(CMAKEJS_ADDON_NAME)".node is online!");
+}
+
+Napi::Value Version(const Napi::CallbackInfo& info) {
+  return Napi::Number::New(info.Env(), NAPI_VERSION);
+}
+
+Napi::Object Init(Napi::Env env, Napi::Object exports) {
+
+  // Export a chosen C++ function under a given Javascript key
+  exports.Set(
+    Napi::String::New(env, "hello"), // Name of function on Javascript side...
+    Napi::Function::New(env, Hello)  // Name of function on C++ side...
+  );
+
+  exports.Set(
+    Napi::String::New(env, "version"),
+    Napi::Function::New(env, Version)
+  );
+
+  // The above will expose the C++ function 'Hello' as a javascript function
+  // named 'hello', etc...
+  return exports;
+}
+
+// Register a new addon with the intializer function defined above
+NODE_API_MODULE(CMAKEJS_ADDON_NAME, Init) // (name to use, initializer to use)
+
+} // namespace NAPI_CPP_CUSTOM_NAMESPACE
+} // namespace Napi
+
+// Export your custom namespace to outside of the Napi namespace, providing an
+// alias to the Napi Addon API; e.g., '<vendor>::<addon>::Object()', along with the
+// functions defined above, such as '<vendor>::<addon>::Hello()'.
+namespace NAPI_CPP_CUSTOM_NAMESPACE::CMAKEJS_ADDON_NAME {
+  using namespace Napi::NAPI_CPP_CUSTOM_NAMESPACE;
+}
```

Comment:
Now I have ```Nathan::Addon::Hello()```.

Let's say I put a second addon on my CMakeLists.txt, with it's own source file. Link the first one to the second one. Now, ```Nathan::Addon::Hello()``` is available in both source files. 

I might even call that function in my second addon and pass one ```Napi::Env``` to another.

Is this possible? I really don't know, haven't tried yet, but I think context-aware addons probably can do this.

Again, it will not break my heart if this line of code was wiped. Just more "stuff folks can do".